sw/airborne/firmwares/rotorcraft/guidance/guidance_v.c

```diff
@@ -481,8 +481,18 @@ static void run_hover_loop(bool in_flight)
 bool guidance_v_set_guided_z(float z)
 {
   if (guidance_v_mode == GUIDANCE_V_MODE_GUIDED) {
+    /* disable vertical velocity setpoints */
     guidance_v_guided_vel_enabled = false;
+
+    /* set altitude setpoint */
     guidance_v_z_sp = POS_BFP_OF_REAL(z);
+
+    /* reset speed setting */
+    guidance_v_zd_sp = 0;
+
+    /* reset guidance reference */
+    guidance_v_z_sum_err = 0;
+    GuidanceVSetRef(guidance_v_z_sp, guidance_v_zd_sp, 0);
```

Comment:
Is it possible to add this to the init function in this file? For the IMAV we are using the flight plan in combination with guided commands for autonomous flight, we startup in guided mode so don't switch to it...